opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -741,11 +736,26 @@ DenseMatrix transposeDenseMatrix(const DenseMatrix& M)
             if (recreate_solver || !flexibleSolver_) {
```

Comment:
recreate_solver appears to be uninitialized due to the move to `shouldCreateSolver`. Hence either the constructor needs to initialize `recreate_solver=shouldCreateSolver()` or the method needs to be used here. Personally, I prefer the former as the latter involves string processing.

Reply:
I think you were confused by the refactoring of ISTLSolverFlexible. There is no shouldCreateSolver() in this class, and recreate_solver is unchanged. I want to make that refactoring here as well, but held back so as to not create more reviewing mess than necessary. Sorry for causing confusion!

I do not understand the string processing comment?

Reply:
You are right. I am confused and that code was only removed from ISTLSolverFlexibleEbos.hpp.